src/main/java/org/broadinstitute/hellbender/utils/nio/PathLineIterator.java

```diff
@@ -0,0 +1,60 @@
+package org.broadinstitute.hellbender.utils.nio;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Iterator;
+import java.util.Spliterator;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.Utils;
+
+/**
+ * Iterate through the lines of a Path. Works for anything you can point
+ * a Path to, such as a normal file or anything you have an NIO provider
+ * for (e.g. GCS).
+ */
+public class PathLineIterator implements AutoCloseable, Iterable<String> {
+    private final Stream<String> lines;
+
+    /**
+     * Returns an iterator so you can iterate over the lines in the text file like so:
+     * for (String line: new Utils.LineIterator(path)) {
```

Comment:
Usage example needs updating to reflect the new class name.

Reply:
Done.